### PR TITLE
UnixPB: Add Vagrant Playbook Checks On Ubuntu 22.04

### LINKED ISSUE #2697: VPC - Upgrade Ubuntu2104 to Ubuntu2204
The VPC for ubuntu 2104 keeps failing https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/OS=Ubuntu2104,label=vagrant/1500/console

```
12:52:44 TASK [Common : Install gnupg2 for aarch64] *************************************
12:52:44 skipping: [127.0.0.1]
12:52:44 
12:52:44 TASK [Common : Add the openjdk repository to apt] ******************************
12:53:10 fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "apt cache update failed"}
12:53:10 
12:53:10 PLAY RECAP *********************************************************************
12:53:10 127.0.0.1                  : ok=8    changed=1    unreachable=0    failed=1    skipped=23   rescued=0    ignored=0   
```

We should look at upgrading this box to Ubuntu 22.04 LTS

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #2697 

This PR requires a change to the build scripts to mirror the inclusion of the LIBRARY_PATH variable to ensure autoconf/configure works as specified.

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)

### CODE REVIEW

**Comment by steelhead31** [comment]
VPC Run For Ubuntu 22.04 in progress here: https://ci.adoptium.net/job/VagrantPlaybookCheck/OS=Ubuntu2204,label=vagrant/1811/console
VPC Run For Ubuntu 20.04 in progress here: 
https://ci.adoptium.net/job/VagrantPlaybookCheck/OS=Ubuntu2004,label=vagrant/1811/console

Depends on https://github.com/adoptium/temurin-build/pull/3548 for the build on Ubuntu 22.04 to succeed

File: ansible/pbTestScripts/vmDestroy.sh [hunk 1]
```diff
@@ -66,7 +68,7 @@ checkOS() {
   	"Windows2022" | "Win2022" | "W22" | "w22" )
 	                       osToDestroy="W2022";;
 	              "all" )
-                        osToDestroy="U16 U18 U20 U21 C6 C7 C8 D8 D10 FBSD12 Sol10 W2012" ;;
+                        osToDestroy="U16 U18 U20 U21 U22 6 C7 C8 D8 D10 FBSD12 Sol10 W2012 W2022" ;;
```
Comment (sxa) [inline ansible/pbTestScripts/vmDestroy.sh:71]:
```suggestion
                        osToDestroy="U16 U18 U20 U21 U22 C6 C7 C8 D8 D10 FBSD12 Sol10 W2012 W2022" ;;
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml [hunk 2]
```diff
@@ -85,10 +85,19 @@
...
-- name: Running ./configure & make for nasm
-  shell: cd /tmp/nasm-2.13.03 && CC={{ CC }} ./configure -prefix=/usr/local && make install
+- name: Running ./configure & make for nasm ( Not Ubuntu 22+ )
+  shell: cd /tmp/nasm-2.13.03 && CC={{ CC }} && ./configure -prefix=/usr/local && make install
   when:
-    - (nasm_installed.rc is defined) and ((nasm_installed.rc != 0 ) or (nasm_installed.rc == 0 and nasm.stdout is version_compare('2.13.03', operator='ne')) )
+    - (ansible_distribution != "Ubuntu" or ansible_distribution == "Ubuntu" and ansible_distribution_major_version < "22") and (nasm_installed.rc is defined) and ((nasm_installed.rc != 0 ) or (nasm_installed.rc == 0 and nasm.stdout is version_compare('2.13.03', operator='ne')) )
+  tags: nasm
+
+- name: Running ./configure & make for nasm ( Ubuntu 22+ x64 )
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml:94]: Hmmm so this will do nothing on Ubuntu 22.04 on non-x64?

**Review by sxa — CHANGES_REQUESTED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml (hunk 2 above)
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml:94]: Yup, Im currently restricting it to where the problem has been seen.. I'll try and do a test on Arm64 Ubuntu 22 before merging.

File: ansible/pbTestScripts/vmDestroy.sh [hunk 3]
```diff
@@ -47,6 +47,8 @@ checkOS() {
                         osToDestroy="U20";;
                 "Ubuntu2104" | "U21" | "u21" )
                         osToDestroy="U21";;
+								"Ubuntu2204" | "U22" | "u22" )
+				                        osToDestroy="U22";;
```
Comment (adamfarley) [inline ansible/pbTestScripts/vmDestroy.sh:51]:
```suggestion
                        osToDestroy="U22";;
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml (hunk 2 above)
Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml:94]: Ah, nasm is only installed on x64, so it doesn't impact ARM in this playbook. Im just checking a build on arm64 ubuntu, for the other half of this fix.

Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml:94]: I need to reflect the build change for Arm64 though...

Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/nasm/tasks/main.yml:94]: > Ah, nasm is only installed on x64

Fair point - I missed that ;-) Will approve now

**Review by sxa — APPROVED** [review]
With the reinstated `C6` this LGTM

**Review by karianna — APPROVED** [review]

**Review by adamfarley — COMMENTED** [review]
LGTM

**Pull request merged**